src/components/App/App.js

```diff
@@ -73,38 +204,96 @@ function App() {
+    })
+    .catch((err) => console.log(err));
+  },[]);
+
   return (
     <div className="page">
-      <Header 
-        onLoginButtonClick={handleLoginButtonClick}
-        isLogged={isLogged}
-      />
-      <Switch>
-        <Route exact path="/">
-          <Main search={handleSearchSubmit}/>
-          <Preloader />
-          <ErrorSearch />
-          <NewsCardList cards={cards}/> 
-          <About />         
-        </Route> 
-        <Route exact path="/saved-news">
-          <SavedNewsHeader  
-            name="Elise" 
-            quantitySavedCards={cards.length}
-            strKeywords={strKeywords}
-          /> 
-          <SavedNews cards={cards}/>
-        </Route>         
-      </Switch>
-      <PopupWithForm 
-              name='signin' 
-              title='Sign in' 
-              isOpen={isSigninPopupOpen}
-              onClose={closeAllPopups}
-              buttonText='Sign in'
-              onSubmit={handleLoginFormSubmit}
-      />
-      <Footer />
+      <CurrentUserContext.Provider value={currentUser}>
+        <Header 
+          onLoginButtonClick={handleLoginButtonClick}
+          onLogoutButtonClick={handleLogoutButtonClick}
+          isLogged={isLogged}
+        />
+        <Switch>
+          <Route exact path="/">
+            <Main search={handleSearchSubmit}/>
+            <Preloader 
+              isOpen={isPreloaderOpen}
+            />
+            <ErrorSearch
+              errorText={errorText}
+              isOpen={isErrorSearchOpen}
+            />
+            <NewsCardList 
+              cards={cards}
+              isOpen={isCardListOpen}
+              isLogged={isLogged}
+              onSaveButtonClick={handleSaveButtonClick}
+            /> 
+            <About />         
+          </Route> 
+          <Route exact path="/saved-news">
+            <SavedNewsHeader   
+              quantitySavedCards={savedCards.length}
+              strKeywords={strKeywords}
+            /> 
+            <SavedNews cards={savedCards} onDeleteButtonClick={handleDeleteButtonClick}/>
+          </Route>         
+        </Switch>
+        <PopupWithForm 
+                name='signin' 
+                title='Sign in' 
+                isOpen={isSigninPopupOpen}
+                onClose={closeAllPopups}
+                buttonText='Sign in'
+                onSignupLinkClick={handleSignupLinkClick}
+                login={handleLoginFormSubmit}
+                useForm={useFormWithValidation}
+                isErrorSubmitVisibled = {isErrorSubmitVisibled}
+                errorText = {errorText}
+        />
+        <Register 
+                name='signup' 
+                title='Sign up' 
+                isOpen={isSignupPopupOpen}
+                onClose={closeAllPopups}
+                buttonText='Sign up'
+                onSigninLinkClick={handleSigninLinkClick}
+                register={handleLogupFormSubmit}
+                useForm={useFormWithValidation}
```

Comment:
* you need to put `useFormWithValidation` into a separate folder `hooks` in file `useFormWithValidation.js` and then import to the components that need it. You should not pass it to the props. There is not such need. It make the code more complicated and dependant